### PR TITLE
Support for (sub-)resources

### PR DESCRIPTION
**Pull request by DocSkellington**
Hello,

For a project, I wanted to store strings in a resource file and be able to translate them. So, I added support for resources and sub-resources. That is, now, *.tres (and *.res) files should work.

For instance, the field `name` can be extracted from
```
[resource]
script = ExtResource(1)
name = "Resource name"
```

Thank you for your work!

### CODE REVIEW

File: babel_godot.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 __version__ = '0.2'
 
 
-_godot_node = re.compile(r'^\[node name="([^"]+)" (:?type="([^"]+)")?')
+_godot_node = re.compile(r'^\[(sub_)?resource|^\[node name="([^"]+)" (:?type="([^"]+)")?')
```
Comment (remram44) [inline babel_godot.py]: I think it should be `(?:sub)` no? I guess you used your code only on resource files and not on scenes?

**Review by remram44 — CHANGES_REQUESTED** [review]
Thanks, that sounds like a great idea! Just got to make sure it keeps working on scene files

File: babel_godot.py (hunk 1 above)
Comment (DocSkellington) [reply babel_godot.py]: I tried with both resources and scenes and it worked. I'm not very familiar with regex extensions, though. So, I've added what you asked.

**Comment by remram44** [comment]
I actually made this into a different extractor, so you can use a mapping file like:
```ini
[python: **.gd]
encoding = utf-8
extract_messages = tr

[godot_scene: **.tscn]
encoding = utf-8

[godot_resource: **.tres]
encoding = utf-8
```
and do:
```
pybabel extract -F babelrc -k Label/text -k Resource/fieldname -k tr -o translations.pot .
```

**Pull request merged**